subsys/net/lib/tls_credentials/tls_credentials_shell.c

```diff
@@ -0,0 +1,813 @@
+static int filter_nonprint(char *buf, size_t len, char inval)
+{
+	int i;
+	int ret = 0;
+
+	for (i = 0; i < len; i++) {
+		if (!isprint((int)buf[i])) {
+			buf[i] = inval;
+			ret = -EINVAL;
+		}
+	}
+
+	return ret;
+}
+
+/* Verify that a provided string consists only of the characters 0-9*/
+static bool check_numeric(char *str)
+{
+	int i;
+	int len = strlen(str);
+
+	for (i = 0; i < len; i++) {
+		if (!isdigit((int)str[i])) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+/* Clear the credential write buffer, returns true if anything was actually cleared. */
+static bool cred_buf_clear()
+{
+	bool cleared = cred_written != 0;
+
+	(void)memset(cred_buf, 0, sizeof(cred_buf));
+	cred_written = 0;
+
+	return cleared;
+}
+
+/* Parse a (possibly incomplete) chunk into the credential buffer */
+static int cred_buf_write(char *chunk)
+{
+	char* writehead = cred_buf + cred_written;
+	size_t chunk_len = strlen(chunk);
+
+	/* Verify that there is room for the incoming chunk */
+	if (writehead + chunk_len >= cred_buf + sizeof(cred_buf) - 1) {
+		return -ENOMEM;
+	}
+
+	/* Append chunk to the credential buffer.
+	 * Deliberately do not copy NULL terminator.
+	 */
+	memcpy(writehead, chunk, chunk_len);
+	cred_written += chunk_len;
+
+	return chunk_len;
+}
+
+/* Get the human-readable name of a TLS credential type */
+static char* cred_type_name(enum tls_credential_type type)
+{
+	/* Scan over predefined type strings, and return the name
+	 * of the first one of matching type.
+	 */
+	for (int i = 0; i < NUM_TYPE_STRINGS; i++) {
+		if (type_strings[i].type == type) {
+			return type_strings[i].name;
+		}
+	}
+
+	/* No matches found, it's invalid. */
+	return "INVALID";
+}
+
+/* Get the compact name of a TLS credential type*/
+static char* cred_type_name_compact(enum tls_credential_type type)
+{
+	/* Scan over predefined type strings, and return the name
+	 * of the last one of matching type.
+	 */
+	for (int i = NUM_TYPE_STRINGS - 1; i >= 0; i--) {
+		if (type_strings[i].type == type) {
+			return type_strings[i].name;
+		}
+	}
+
+	/* No matches found, it's invalid. */
+	return "INV";
+}
+
+/* Shell interface routines */
+
+/* Attempt to parse a command line argument into a sectag.
+ * TLS_SEC_TAG_NONE is returned if ANY_KEYWORD is provided.
+ */
+static int shell_parse_cred_sectag(const struct shell *sh, char *arg, sec_tag_t *out,
+				   bool allow_any)
+{
+
+	unsigned long sectag_value;
+	int err = 0;
+
+	/* Check for "ANY" special keyword if desired. */
+	if (allow_any && strcasecmp(arg, ANY_KEYWORD) == 0) {
+		*out = TLS_SEC_TAG_NONE;
+		return 0;
+	}
+
+	/* Otherwise, verify that the sectag is purely numeric */
+	if (!check_numeric(arg)) {
+		err = -EINVAL;
+		goto error;
+	}
+
+	/* Use strtoul because it has nicer validation features than atoi */
+	sectag_value = shell_strtoul(arg, 10, &err);
+
+	if (!err) {
+		*out = (sec_tag_t)sectag_value;
+		return 0;
+	}
+
+error:
+	shell_fprintf(
+		sh, SHELL_ERROR,
+		"%s is not a valid sectag.\n", arg
+	);
```

Comment:
TBH I'm not a fan of breaking the lines when not really needed, this could easily fit into a single line, looks kind of bloated right now.